### ISSUE TITLE
wrong file permissions in gem (tar)

### ISSUE THREAD

**Issue by tompata** [nfm/gplus#4]
in the gem data.tar.gz, where all the (lib) files are located, there's wrong file permissions.
there's only RW permissions for the owner, but all the other read permissions are missing! (0600)

that's why in our server the gplus gem is unavailable: uninitialized constant Gplus

the expected, correct file permissions would be:
0644

please modify the file permissions before build the next gem version!

thanks,
Tamas

**Comment by tompata** [comment]
Hi @nfm 

Would you please fix these file permissions?
It can cause problems in hosting, and the solution is not always easy (because of user permissions)

Thanks
Tamas

**Comment by nfm** [comment]
Hi @tompata,

Sorry for the slow response. I'm having trouble getting this to play nicely. I've updated my development repository to have the correct file permissions:

```
-rw-r--r-- 1 nfm nfm  2247 2012-01-04 09:28 CHANGELOG.md
drwxr-xr-x 3 nfm nfm  4096 2012-01-02 09:57 coverage
drwxr-xr-x 5 nfm nfm  4096 2012-01-04 09:24 doc
-rw-r--r-- 1 nfm nfm    38 2011-09-16 10:31 Gemfile
-rw-r--r-- 1 nfm nfm   945 2011-12-30 18:52 Gemfile.lock
-rw-r--r-- 1 nfm nfm  1121 2012-01-04 09:28 gplus.gemspec
drwxr-xr-x 3 nfm nfm  4096 2012-01-04 09:28 lib
drwxr-xr-x 2 nfm nfm  4096 2012-01-11 07:49 pkg
-rw-r--r-- 1 nfm nfm   275 2012-01-04 09:28 Rakefile
-rw-r--r-- 1 nfm nfm 10866 2012-01-05 09:51 README.md
drwxr-xr-x 4 nfm nfm  4096 2011-11-13 15:34 spec
```

When I build, then manually untar the gem, the correct permissions are preserved.

When I gem install it, only the file permissions are preserved, and all directories are 0700:

```
-rw-r--r-- 1 nfm nfm  2247 2012-01-11 07:49 CHANGELOG.md
-rw-r--r-- 1 nfm nfm    38 2012-01-11 07:49 Gemfile
-rw-r--r-- 1 nfm nfm  1121 2012-01-11 07:49 gplus.gemspec
drwx------ 3 nfm nfm  4096 2012-01-11 07:49 lib
-rw-r--r-- 1 nfm nfm   275 2012-01-11 07:49 Rakefile
-rw-r--r-- 1 nfm nfm 10866 2012-01-11 07:49 README.md
drwx------ 4 nfm nfm  4096 2012-01-11 07:49 spec
```

I'm using rbenv, which might be causing this for me.

What is your production environment running? RVM, rbenv, system install of gems, or installing to vendor/bundle?

I'd like to confirm my permission changes will work before pushing a new version, that could potentially not fix the problem.

**Comment by tompata** [comment]
hi @nfm,

we're using RVM in local (development env), and bundler in production.
how can i check and confirm the changes?

**Comment by nfm** [comment]
Thanks for the extra info.

I've looked into this a bit more, and I think my issue with directory permissions was a umask thing. My local umask is 0077, and the gem directories are 0700. But my production umask is 0022, and gem directories are correctly 0755.

So it seems like file permissions are preserved from the data.tar.gz, but directories are created according to your umask.

Could you confirm your production umask, and whether all your gem directories in production are group and other readable and executable? If so, I think me changing the permissions on the files will be sufficient to fix this issue.

Thanks!

**Comment by tompata** [comment]
Yes, umask in production is usually 022. Gem directories have 0755 permissions.

**Comment by nfm** [comment]
Thanks. I've just released 2.0.1 with world readable permissions on the files. 

Give it a try when you have a chance, and let me know how it goes.

**Comment by tompata** [comment]
It's working! Thank you very much!
